docs/writing-tests/snapshot-testing.md

```diff
@@ -8,53 +8,173 @@ Storybook is a helpful tool for snapshot testing because every story is essentia
+
+#### Setup
+
+We recommend you turn off your current storyshots tests to start the migration process. To do this, rename the configuration file (i.e., `storybook.test.ts|js` or similar) to `storybook.test.ts|js.old`. This will prevent the tests from being detected, as you'll create a new testing configuration file with the same name. By doing this, you'll be able to preserve your existing tests while transitioning to portable stories before removing the Storyshots addon from your project.
+
+#### Import project-level annotations from Storybook
+
+If you need project-level annotations (e.g., [decorators](../writing-stories/decorators.md), [args](../writing-stories/args.md), styles) enabled in your `./storybook/preview.js|ts` included in your tests, adjust your test set up file to import the annotations as follows:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/portable-stories-import-annotations.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<Callout variant="info" icon="ℹ️">
+
+If you're using Vue3, you must install the [`@storybook/testing-vue3`](https://storybook.js.org/addons/@storybook/testing-vue3) package to use the `setProjectAnnotations` API in your setup file and the `composeStories` API in your existing tests.
 
 </Callout>
 
+#### Configure the testing framework for portable stories
+
+To help you migrate from Storyshots addon to Storybook's portable stories with the `composeStories` helper API, we've prepared examples to help you get started. Listed below are examples of two of the most popular testing frameworks: [Jest](https://jestjs.io/) and [Vitest](https://vitest.dev/). We recommend placing the code in a newly created `storybook.test.ts|js` file and adjusting the code accordingly, depending on your testing framework. Both examples below will:
+
+- Import all story files based on a glob pattern
+- Iterate over these files and use `composeStories` on each of their modules, resulting in a list of renderable components from each story
+- Cycle through the stories, render them, and snapshot them
+
+#### Vitest
+
+If you're using [Vitest](https://vitest.dev/) as your testing framework, you can begin migrating your snapshot tests to Storybook's portable stories with the `composeStories` helper API by referring to the following example. You will need to modify the code in your `storybook.test.ts|js` file as follows:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/portable-stories-vitest-snapshot-test.js.mdx',
+    'common/portable-stories-vitest-snapshot-test.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+When your test is executed with Vitest, it will generate a single snapshot file (i.e., `storybook.test.ts|js.snap`) with all the stories in your project. However, if you want to generate individual snapshot files, you can use Vitest's [`toMatchFileSnapshot`](https://vitest.dev/guide/snapshot.html#file-snapshots) API. For example:
```

Comment:
```suggestion
After placing the contents of the file (and adapting it as needed), run your test command to try it out. When your Storybook test file is executed with Vitest, it will generate a single snapshot file (i.e., `storybook.test.ts|js.snap`) with all the stories in your project. However, if you want to generate individual snapshot files, you can use Vitest's [`toMatchFileSnapshot`](https://vitest.dev/guide/snapshot.html#file-snapshots) API. For example:
```